src/shared/particle_dynamics/execution_unit/execution_context.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef SPHINXSYS_EXECUTION_CONTEXT_HPP
+#define SPHINXSYS_EXECUTION_CONTEXT_HPP
+
+#include <sycl/sycl.hpp>
+
+namespace SPH {
+    namespace execution {
+        class ParallelSYCLDevicePolicy;
+
+        template<typename ExecutionUnit>
+        struct Context {
+            template<class ...Args>
+            explicit Context(Args& ...args) {};
+
+            template<class ...Args>
+            explicit Context(Args&& ...args) {};
```

Comment:
Superfluous ?

Reply:
The assert could be put anywhere inside the class if it applies on the class template arguments instead ?